src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -59,6 +61,47 @@ std::string MutationReplaceBinaryOperator::GetExpr(
   return result;
 }
 
+bool MutationReplaceBinaryOperator::IsRedundantReplacementOperator(
+    clang::BinaryOperatorKind op, clang::ASTContext& ast_context) const {
+  clang::Expr::EvalResult eval_result;
+  bool rhs_is_int =
+      binary_operator_.getRHS()->EvaluateAsInt(eval_result, ast_context);
+
+  // In the following cases, the replacement is equivalent to either replacement
+  // with a constant or argument replacement.
+  // See https://github.com/mc-imperial/dredd/issues/89 for a more in-depth
+  // explanation.
+  if (rhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                              llvm::APSInt::get(0))) {
+    if (op == clang::BO_Add || op == clang::BO_Sub || op == clang::BO_Shl ||
+        op == clang::BO_Shr || op == clang::BO_Mul || op == clang::BO_Rem) {
```

Comment:
I am not sure about the "rem" case - remainder by zero is an undefined behaviour so it's not covered by other cases.

Reply:
Didn't you note that the undefined behavior case was covered by `/`?

Reply:
Ah yes - you're right. When you add comments it would be good to note that.